ember-client/shared/src/main/scala/org/http4s/ember/client/EmberClientBuilder.scala

```diff
@@ -253,20 +254,16 @@ final class EmberClientBuilder[F[_]: Async] private (
           .withMaxTotal(maxTotal)
           .withOnReaperException(_ => Applicative[F].unit)
       pool <- builder.build
-      optH2 <- (Alternative[Option].guard(enableHttp2) >> tlsContextOptWithDefault).traverse(
-        context =>
-          H2Client.impl[F](
-            pushPromiseSupport.getOrElse { case (_, _) => Applicative[F].pure(Outcome.canceled) },
-            context,
-            logger,
-            if (pushPromiseSupport.isDefined) default
-            else {
-              default.copy(enablePush =
-                org.http4s.ember.core.h2.H2Frame.Settings.SettingsEnablePush(false)
-              )
-            },
-          )
-      )
+      optH2 <- (if (enableHttp2) tlsContextOptWithDefault else None).traverse { context =>
```

Comment:
Is it the case when `Applicative#when` fits well?

Reply:
No, because this yields an `Option`, and the `when` is for `F[Unit]`. In any case, that is still less straightforward than `if-then-else`.